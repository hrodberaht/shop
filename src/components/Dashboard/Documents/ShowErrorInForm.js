import React from 'react';
import PropTypes from 'prop-types';

const ShowErrorInForm = ({ className, touched, error }) => (
  <p className={className}>{touched && (error && <span>{error}</span>)}</p>
);

ShowErrorInForm.defaultProps = {
  error: null,
  touched: false,
  className: null
};

ShowErrorInForm.propTypes = {
  touched: PropTypes.bool,
  error: PropTypes.string,
  className: PropTypes.string
};

export default ShowErrorInForm;
