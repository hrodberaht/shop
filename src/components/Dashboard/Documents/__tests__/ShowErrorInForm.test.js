import ShowErrorInForm from '../ShowErrorInForm';

const defaultProps = {
  touched: false,
  error: null,
  className: 'btn'
};

const setup = buildSetup(ShowErrorInForm, defaultProps);

describe('<ShowErrorInForm />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('when touched should show error', () => {
    const { wrapper } = setup();
    wrapper.setProps({ touched: true, error: 'error' });
    expect(wrapper).toMatchSnapshot();
  });
});
