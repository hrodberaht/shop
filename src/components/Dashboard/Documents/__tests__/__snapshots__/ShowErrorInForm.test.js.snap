// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowErrorInForm /> should render without crash 1`] = `
<p
  className="btn"
/>
`;

exports[`<ShowErrorInForm /> when touched should show error 1`] = `
<p
  className="btn"
>
  <span>
    error
  </span>
</p>
`;
